### ISSUE TITLE
Ability to use SOCK_DGRAM sockets

### ISSUE THREAD

**Issue by VidarHUN** [envoyproxy/envoy#11239]
I want to route UDP traffic to a Unix Domain Socket, but I can't do that right now because there is no way to open the socket in SOCK_DGRAM mode. I think it would be a very useful feature if it cloud do that. Is there any chance that you will be able to support it in the near future?

**Comment by mattklein123** [comment]
Yeah should be not too hard to add this behavior. Will mark help wanted.

**Comment by VidarHUN** [comment]
Thanks a lot!
